test/dates/ranges.jl

```diff
@@ -503,3 +499,42 @@ let d = Dates.Day(1)
     @test (Dates.Date(2000):d:Dates.Date(2001))+d == (Dates.Date(2000)+d:d:Dates.Date(2001)+d)
     @test (Dates.Date(2000):d:Dates.Date(2001))-d == (Dates.Date(2000)-d:d:Dates.Date(2001)-d)
 end
+
+# Time ranges
+dr  = Dates.Time(23,1,1):Dates.Time(23,2,1)
+dr1 = Dates.Time(23,1,1):Dates.Time(23,1,1)
+dr2 = Dates.Time(23,1,1):Dates.Time(22,2,1) # empty range
+dr3 = Dates.Time(23,1,1):Dates.Minute(-1):Dates.Time(22,1,1) # negative step
+# Big ranges
+dr8 = typemin(Dates.Time):typemax(Dates.Time)
+dr9 = typemin(Dates.Time):Dates.Nanosecond(1):typemax(Dates.Time)
+# Non-default steps
+dr10 = typemax(Dates.Time):Dates.Microsecond(-1):typemin(Dates.Time)
+dr11 = typemin(Dates.Time):Dates.Millisecond(1):typemax(Dates.Time)
+dr12 = typemin(Dates.Time):Dates.Minute(1):typemax(Dates.Time)
+dr13 = typemin(Dates.Time):Dates.Hour(1):typemax(Dates.Time)
+dr14 = typemin(Dates.Time):Dates.Millisecond(10):typemax(Dates.Time)
+dr15 = typemin(Dates.Time):Dates.Minute(100):typemax(Dates.Time)
+dr16 = typemin(Dates.Time):Dates.Hour(1000):typemax(Dates.Time)
+dr17 = typemax(Dates.Time):Dates.Millisecond(-10000):typemin(Dates.Time)
+dr18 = typemax(Dates.Time):Dates.Minute(-100):typemin(Dates.Time)
+dr19 = typemax(Dates.Time):Dates.Hour(-10):typemin(Dates.Time)
+dr20 = typemin(Dates.Time):Dates.Microsecond(2):typemax(Dates.Time)
+
+drs = Any[dr,dr1,dr2,dr3,dr8,dr9,dr10,
+          dr11,dr12,dr13,dr14,dr15,dr16,dr17,dr18,dr19,dr20]
+
+@test map(length,drs) == map(x->size(x)[1],drs)
+@test all(x->findin(x,x) == [1:length(x);], drs[1:4])
+@test isempty(dr2)
+@test all(x->reverse(x) == last(x):-step(x):first(x),drs)
+@test all(x->minimum(x) == (step(x) < zero(step(x)) ? last(x) : first(x)),drs[4:end])
+@test all(x->maximum(x) == (step(x) < zero(step(x)) ? first(x) : last(x)),drs[4:end])
+@test_throws MethodError dr + 1
+
+a = Dates.Time(23,1,1)
+@test map(x->a in x,drs[1:4]) == [true,true,false,true]
+@test a in dr
+
+@test all(x->sort(x) == (step(x) < zero(step(x)) ? reverse(x) : x),drs)
+@test all(x->step(x) < zero(step(x)) ? issorted(reverse(x)) : issorted(x),drs)
```

Comment:
newly added lines aren't going to conflict with the other PR if it won't need to touch them